File: app/src/main/java/com/gpacini/daysuntil/data/model/Event.java
package com.gpacini.daysuntil.data.model;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Created by gavinpacini on 10/10/15.
 */
public class Event implements Parcelable {

    private String uuid;
    private String title;
    private long timestamp;

    public String getUuid() {
        return uuid;
    }

    public void setUuid(String uuid) {
        this.uuid = uuid;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public long getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(long timestamp) {
        this.timestamp = timestamp;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.uuid);
        dest.writeString(this.title);
        dest.writeLong(this.timestamp);
    }

    public Event(RealmEvent realmEvent) {
        this.uuid = realmEvent.getUuid();
        this.title = realmEvent.getTitle();
        this.timestamp = realmEvent.getTimestamp();
    }

    protected Event(Parcel in) {
        this.uuid = in.readString();
        this.title = in.readString();
        this.timestamp = in.readLong();
    }

    public static final Parcelable.Creator<Event> CREATOR = new Parcelable.Creator<Event>() {
        public Event createFromParcel(Parcel source) {
            return new Event(source);
        }

        public Event[] newArray(int size) {
            return new Event[size];
        }
    };


    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Event event = (Event) o;

        if (timestamp != event.timestamp) return false;
        if (!uuid.equals(event.uuid)) return false;
        return title.equals(event.title);

    }

    @Override
    public int hashCode() {
        int result = uuid.hashCode();
        result = 31 * result + title.hashCode();
        result = 31 * result + (int) (timestamp ^ (timestamp >>> 32));
        return result;
    }
}

File: app/src/main/java/com/gpacini/daysuntil/data/model/Event.kt
package com.gpacini.daysuntil.data.model

import android.os.Parcel
import android.os.Parcelable

/**
 * Created by gavinpacini on 10/10/15.
 */
class Event : Parcelable {

    public var uuid: String? = null
    public var title: String? = null
    public var timestamp: Long = 0

    override fun describeContents(): Int = 0

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(this.uuid)
        dest.writeString(this.title)
        dest.writeLong(this.timestamp)
    }

    constructor(realmEvent: RealmEvent) {
        this.uuid = realmEvent.uuid
        this.title = realmEvent.title
        this.timestamp = realmEvent.timestamp
    }

    protected constructor(input: Parcel) {
        this.uuid = input.readString()
        this.title = input.readString()
        this.timestamp = input.readLong()
    }


    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false

        val event = o as Event?

        if (timestamp != event?.timestamp) return false
        if (uuid != event?.uuid) return false
        return title == event?.title

    }

    override fun hashCode(): Int {
        var result = uuid!!.hashCode()
        result = 31 * result + title!!.hashCode()
        result = 31 * result + (timestamp xor (timestamp.ushr(32))).toInt()
        return result
    }

    companion object {

        val CREATOR: Parcelable.Creator<Event> = object : Parcelable.Creator<Event> {
            override fun createFromParcel(source: Parcel): Event {
                return Event(source)
            }

            override fun newArray(size: Int): Array<Event?> {
                return arrayOfNulls(size)
            }
        }
    }
}
